tests/test.websock.js

```diff
@@ -185,61 +185,48 @@ describe('Websock', function () {
                 expect(sock.rQi).to.equal(5);
             });
         });
+    });
 
-        describe('flush', function () {
-            beforeEach(function () {
-                sock._websocket = {
-                    send: sinon.spy()
-                };
-            });
+    describe('Send queue methods', function () {
+        let sock;
 
+        beforeEach(function () {
+            let websock = new FakeWebSocket();
+            websock._open();
+            sock = new Websock();
+            sock.attach(websock);
+        });
+
+        describe('flush', function () {
             it('should actually send on the websocket', function () {
-                sock._websocket.bufferedAmount = 8;
-                sock._websocket.readyState = WebSocket.OPEN;
                 sock._sQ = new Uint8Array([1, 2, 3]);
                 sock._sQlen = 3;
                 const encoded = sock._encodeMessage();
```

Comment:
From the variable & function names, it seems we're still looking at internal stuff here? If we can't avoid that, perhaps we can add a comment, like you did in other commits.

Reply:
Are you referring to `_sQ` and `_sQlen`? Those are unfortunately the current API. :/

I have another branch soon incoming which will clean those up and add a more clean API on the send side as well.